amethyst_renderer/src/sprite.rs

```diff
@@ -61,6 +68,141 @@ impl From<[f32; 4]> for Sprite {
+                position: [half_width, -half_height, 0.0],
+                tex_coord: [1.0, 0.0],
+            },
+            PosTex {
+                position: [-half_width, half_height, 0.0],
+                tex_coord: [0.0, 1.0],
+            },
+            // Second triangle
+            PosTex {
+                position: [half_width, half_height, 0.0],
+                tex_coord: [1.0, 1.0],
+            },
+            PosTex {
+                position: [-half_width, half_height, 0.0],
+                tex_coord: [0.0, 1.0],
+            },
+            PosTex {
+                position: [half_width, -half_height, 0.0],
+                tex_coord: [1.0, 0.0],
+            },
+        ];
+
+        let mesh = self.loader
+            .load_from_data(vertices.into(), (), &self.asset_storage);
+
+        let material = Material {
+            albedo: texture,
+            albedo_offset: TextureOffset {
+                u: (sprite.left / size.0, sprite.right / size.0),
+                v: (1.0 - sprite.bottom / size.1, 1.0 - sprite.top / size.1),
+            },
+            ..self.material_defaults.0.clone()
+        };
+
+        (mesh, material)
+    }
+
+    /// Adds a mesh and a material to an entity corresponding to the sprite and texture given.
+    /// Note that is you need to insert the same sprite and texture, using ``add_multiple`` allows for better performances.
+    pub fn add(
+        &mut self,
+        entity: Entity,
+        sprite: &Sprite,
+        texture: TextureHandle,
+        texture_size: (f32, f32),
+    ) -> Result<()> {
+        let (mesh, material) = self.build_mesh_and_material(sprite, texture, texture_size);
+        self.meshes.insert(entity, mesh)?;
+        self.materials.insert(entity, material)?;
+        Ok(())
+    }
+
+    /// Adds the same mesh and material to multiple entities corresponding to the sprite and texture given.
+    pub fn add_multiple(
+        &mut self,
+        entities: Vec<Entity>,
+        sprite: &Sprite,
+        texture: TextureHandle,
+        texture_size: (f32, f32),
+    ) -> Result<()> {
+        let len = entities.len();
+        if len != 0 {
+            let (mesh, material) = self.build_mesh_and_material(sprite, texture, texture_size);
+            for entity in 0..len - 1 {
+                self.meshes.insert(entities[entity], mesh.clone())?;
+                self.materials.insert(entities[entity], material.clone())?;
+            }
+            self.meshes.insert(entities[len - 1], mesh)?;
+            self.materials.insert(entities[len - 1], material)?;
+        }
+        Ok(())
+    }
+}
+
+/// An easy way to attach and display a sprite when building an entity
+pub trait WithSpriteRender
+where
+    Self: Sized,
+{
+    /// Adds a mesh and a material to the entity being built corresponding to the sprite and texture given.
+    fn with_sprite(
+        self,
+        sprite: &Sprite,
+        texture: TextureHandle,
+        texture_size: (f32, f32),
```

Comment:
There isn't any way to delay the texture_size evaluation (where it is needed) until we check if the texture is loaded in the storage?

Reply:
I couldn't find any unfortunately. All of them include having too much runtime. In case the user does not know the size of the texture at compile time, they should load the texture, get its size thanks to the method I added, and pass it to ``with_sprite``. It's not that bad.

Reply:
Maybe we could have SpriteMesh asset that once loaded can be unwrapped and added to the entity. But that's not less work on the user, it's just different. And it also adds more API which is already a lot for such a simple task.